test/unit/model/helpers/downloader-test.js

```diff
@@ -271,4 +272,31 @@ describe('Downloader', function() {
       expect(successHandler).to.be.calledOnce;
     });
   });
+
+  describe('restartDownload', function(){
+    let options = 'http://example.com/jdk.zip';
+
+    it('should change downloader status from \'Download Failed\' to \'Downloading\'', function(){
+      expect(downloader.restartDownload()).to.be.equal(undefined);
```

Comment:
restartDownload() function does not have 'return' statement in it. This means the function always returns 'undefined' unless exception occurs. There is no need to have this expect call here. 

In test you should create downloader instance first and simulate download error. See example on [lines 205 - 215](https://github.com/redhat-developer-tooling/developer-platform-install/pull/476/files#diff-762967bfd2ff9393807ffee1c812a190R205) except you don't need to stub errorHandler method.
After that downloader will have downloads Map instance populated with url for which download was failed.

Status change for downloader is should be done through progress.setStatus('Downloading') call, so you need to add a spy for fakeProgress object. 

After all preparations above you ready to call the method under test and verify that 'downloader.downloadSize', 'downloader.received', 'downloader.currentSize' were changed to 0 and setStatus spy called with 'Downloading'.